addons/account/models/account_move.py

```diff
@@ -2923,6 +2930,7 @@ def _compute_hash(self, previous_hash):
         """ Computes the hash of the browse_record given as self, based on the hash
         of the previous record in the company's securisation sequence given as parameter"""
         self.ensure_one()
+        self._compute_string_to_hash()  # Force recompute bevause the hash version may have changed
```

Comment:
What? Why? When?

Reply:
When generating the integrity report, we will first use hash v1, and compute a `string_to_hash`. However, if/when we switch to hash v2, since `string_to_hash` has already been computed once, it will not be computed again and it will be wrong because it does not use the new fields in v2.
I'll move it in the integrity check instead

Reply:
If I understand your issue, it would make more sense to invalidate the cache for this field in the test

Reply:
Indeed, but in the integrity report, no ? Why in the test ?

Reply:
In the report is fine too, but there is no reason to have a value in the cache in a real use case

Reply:
Why not ?
If we have multiple moves where the first part was hashed using v1, and a second part using v2, then when switching from v1 to v2, the move will have the v1 `string_to_hash` in the cache, therefore we must at that moment invalidate the cache, i.e. in the loop, to get the v2 `string_to_hash`.